service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java

```diff
@@ -87,6 +93,13 @@ private void assertC2BundleDocument(C2DocumentBundle documentBundle, String desc
         assertThat(documentBundle.getDocument().getFilename()).isEqualTo(document.originalDocumentName);
         assertThat(documentBundle.getDocument().getBinaryUrl()).isEqualTo(document.links.binary.href);
         assertThat(documentBundle.getDescription()).isEqualTo(description);
+
+        List<SupportingEvidenceBundle> supportingBundle = unwrapElements(documentBundle.getSupportingEvidenceBundle());
+        if (!supportingBundle.isEmpty()) {
```

Comment:
What if the returned data is empty? 
What if I break the code and don't return the bundle, whatever test is using that, it's still oing to pass.

Reply:
First test tests supportingEvidenceBundle along with other c2 related data. 

Second test is using different test data to test additional C2 documents and supportingBundle can be empty. 

Seems okay to me.. Let me know what you think.

Reply:
The problem is you can break the code and this code is still passing. I commented out:

Line 171 in the UploadC2DocumentsController:
//            .supportingEvidenceBundle(wrapElements(updatedSupportingEvidenceBundle));

This test is not enforcing the value to be there, it passes even if nothing is returned. That's for sure wrong.

Reply:
Sorry, I am not sure if i understand why test should enforce value to be there. SupportingEvidence can be empty. 

I have updated test now and added supportingEvidence to second test too and removed if condition in the test.. not sure if this is any better!